src/datadog/azext_datadog/generated/custom.py

```diff
@@ -90,19 +91,29 @@ def datadog_monitor_create(client,
                            user_info=None,
                            sku_name=None,
                            no_wait=False):
-    return sdk_no_wait(no_wait,
-                       client.begin_create,
-                       resource_group_name=resource_group_name,
-                       monitor_name=monitor_name,
-                       tags=tags,
-                       location=location,
-                       type=identity_type,
-                       provisioning_state=provisioning_state,
-                       monitoring_status=monitoring_status,
-                       marketplace_subscription_status=marketplace_subscription_status,
-                       datadog_organization_properties=datadog_organization_properties,
-                       user_info=user_info,
-                       name=sku_name)
+    poller = sdk_no_wait(no_wait,
+                         client.begin_create,
+                         resource_group_name=resource_group_name,
+                         monitor_name=monitor_name,
+                         tags=tags,
+                         location=location,
+                         type=identity_type,
+                         provisioning_state=provisioning_state,
+                         monitoring_status=monitoring_status,
+                         marketplace_subscription_status=marketplace_subscription_status,
+                         datadog_organization_properties=datadog_organization_properties,
+                         user_info=user_info,
+                         name=sku_name)
+    result = poller.result()
+    if result and result.principal_id:
+        scrope = '/subscriptions/' + result.id.split('/')[2]
+        print(scrope)
+        get_default_cli().invoke(['role', 'assignment', 'create', '-o', 'none',
+                                  '--assignee-object-id', result.principal_id,
+                                  '--assignee-principal-type', 'ServicePrincipal',
+                                  '--role', 'Monitoring Reader',
+                                  '--scope', scrope])
```

Comment:
Can you call the `create_role_assignment` method from role module directly? We have an example:
https://github.com/Azure/azure-cli-extensions/blob/1b89a454b8d6b8cf216cb21fcbf7226a493e48fb/src/blueprint/azext_blueprint/custom.py#L546-L550

Reply:
Updated. Thanks.